src/mapgen.cpp

```diff
@@ -3423,6 +3433,10 @@ bool mapgen_function_json_base::setup_common( const JsonObject &jo )
             parray.throw_error( string_format( "format: rows: must have at least %d rows, not %d",
                                                expected_dim.y, parray.size() ) );
         }
+        if( static_cast<int>( parray.size() ) != total_size.y ) {
+            parray.throw_error( string_format( "format: rows: must have %d rows, not %d",
```

Comment:
```suggestion
            parray.throw_error( string_format( "format: rows: must have %d rows, not %d; check mapgensize if applicable",
```
Better would be `rows (from mapgensize),`, but it apparently isn't always from mapgensize?

Reply:
It's only from `mapgensize` in the case of nests.  Sure, the extra clarification is probably helpful.